v2/audit/conn/hang_conn.go

```diff
@@ -0,0 +1,46 @@
+package conn
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/microsoft/go-otel-audit/v2/audit/msgs"
+)
+
+// TestHangConn is a test connection that hangs on CloseSend to demonstrate a hanging bug.
+type TestHangConn struct {
+	hangOnCloseSend bool
+}
+
+// NewTestHangConn creates a new test connection that hangs on CloseSend.
+func NewTestHangConn() TestHangConn {
+	if !testing.Testing() {
+		panic("NewTestHangConn should only be used in tests")
+	}
+	return TestHangConn{hangOnCloseSend: true}
+}
+
+// Type returns the connection type for TestHangConn.
+func (t TestHangConn) Type() Type {
+	return TypeTCP
+}
+
+// Write writes a message to the connection. This is a no-op for TestHangConn.
+func (t TestHangConn) Write(ctx context.Context, msg msgs.Msg) error {
+	// Process writes normally
```

Comment:
minor, nit, feel free to ignore: This comment confused me - is normally "like a real connection" (which is not what this is doing) or is normally "don't simulate an error"?

Possibly just remove comment since you already said it's a no-op for TestHangConn, which is more clearly phrased than this.